File: programmers/binary_search/immigration_inspection.py
def solution(n, times):
    l = 0
    r = 10 ** 15
    while (l < r):
        m = (l + r) // 2
        sumt = 0
        for x in times:
            sumt += m // x

        if sumt >= n:
            r = m
        else:
            l = m + 1
    
    answer = l
    return answer

File: programmers/binary_search/immigration_inspection.java
class Solution {
    public long solution(int n, int[] times) {
        long answer = 0;
        
        long l = 0;
        long r = (long) 1e18;
        while (l < r) {
            long mid = (l + r) / 2;
            long cnt = 0;
            for (int x: times) {
                cnt += mid / x;
            }
            if (cnt >= (long) n) r = mid;
            else l = mid + 1;
        }
        
        answer = l;
        
        return answer;
    }
}